VeChainCoreTest/RLPTest.cs

```diff
@@ -32,8 +32,12 @@ public void RLPHexParser()
             var rlpTransaction = new RlpTransaction(realTransaction).AsRlpValues();
 
             var vetEncoded = RlpEncoder.Encode(rlpTransaction);
-           
+
             var out1 = vetEncoded.ByteArrayToString(StringType.Hex | StringType.ZeroLowerX);
+
+            // TODO: what should the encoded bytes of the RlpTransaction match? is this right?
+            Assert.Equal("0xf85e018800000000aabbccdd20f840df947567d83b7b8d80addcb281a71d54fc7b3364ffed82271086000000606060df947567d83b7b8d80addcb281a71d54fc7b3364ffed824e2086000000606060818082520880883132333435363738c1c0", out1);
```

Comment:
RensR, what should this be?

Reply:
@RensR - I pulled this data from the thor sdk's transaction_test.go
(where 0x2a1c25ce0d66f45276a5f308b99bf410e2fc7d5b6ea37a49f2ab9f1da9446478 is the signing hash)
by injecting the following and extracting via debugger;
```go
	bytes, _ := rlp.EncodeToBytes(trx)
	hexBytes := hex.EncodeToString(bytes)
```

Reply:
Actually, the signing hashed bytes should be `0xf8550184aabbccdd20f840df947567d83b7b8d80addcb281a71d54fc7b3364ffed82271086000000606060df947567d83b7b8d80addcb281a71d54fc7b3364ffed824e208600000060606081808252088083bc614ec0` and the full RLP is `0xf8550184aabbccdd20f840df947567d83b7b8d80addcb281a71d54fc7b3364ffed82271086000000606060df947567d83b7b8d80addcb281a71d54fc7b3364ffed824e208600000060606081808252088083bc614ec080` (note the 0x80 at the end).

Reply:
Ah yes, I noticed that too when comparing with the Java SDK. I have not yet found the time to look at the small difference but it would seem it's in the last item. If I remember correctly it might have been something with empty list/null item encoding, but I'm not sure.

Reply:
*edited it again, had it backwards I think

Reply:
yeah I think the full RLP includes an empty list at the end representing an empty signature

Reply:
From VechainCore:
0x prefix
f85e list
01 chaintag
8800000000aabbccdd blockref
20 expiration
f840df947567d83b7b8d80addcb281a71d54fc7b3364ffed82271086000000606060df947567d83b7b8d80addcb281a71d54fc7b3364ffed824e2086000000606060 clauses
8180 gaspricecoef
825208 gas
80 dependson
883132333435363738 nonce
c1c0 reserved

Adding the empty statement would be easy. The first wrong bytes are from the list length, they should be solved when the other values are correct.

Reply:
I have a branch made that uses the Nethereum.RLP instead of the RLP implementation here that produces the right data to satisfy the test - pretty extreme changes though

Reply:
I fixed the nonce encoding, see last commit

Reply:
Now it's just the blockref that is wrong, I'll look into it tomorrow but it should be easily solved now that all the other fields are correct.

Reply:
Cool

Reply:
The test now passes since #7